frontend/packages/integration-tests-cypress/support/selectors.ts

```diff
@@ -19,3 +23,13 @@ Cypress.Commands.add('byTestActionID', (selector: string) =>
 Cypress.Commands.add('byLegacyTestID', (selector: string) =>
   cy.get(`[data-test-id="${selector}"]`),
 );
+Cypress.Commands.add('byButtonText', (selector: string) =>
+  cy.get('button[type="button"]').contains(`${selector}`),
+);
+Cypress.Commands.add('byDataID', (selector: string) => cy.get(`[data-id="${selector}"]`));
+
+Cypress.Commands.add('byTestSelector', (selector: string) => cy.get(`[data-test-selector="${selector}"]`));
+
+Cypress.Commands.add('byTestDropDownMenu', (selector: string) => 
+  cy.get(`[data-test-dropdown-menu="${selector}"]`),
+);
```

Comment:
Why are these extra test attributes needed?  Couldn't they all just be `data-test`?  Do `data-id`, `data-test-selector`, and `data-test-dropdown-menu` already exist in the code base?  Will these `data-` selectors eventually be replaced with `data-test`?

Reply:
Developers have used a different set of attributes like data-test-selector, data-id and there are few buttons which don't have any data-test attribute so we can identify them using buttonText
Instead of updating the dev-code base, I thought to use this type of functions to work with our scripts

Reply:
I recommend updating the dev-code base.  We are trying to standardize on `data-test` due to it's integration with [Cypress Selector Playground](https://docs.cypress.io/guides/core-concepts/test-runner.html#Uniqueness).  Any elements which do not have a data-test attribute should be given `data-test`.  We are supporting the older `data-test-id` attribute via `cy.byLegacyTestID(...)`.  Relying on button text isn't best practice, as I mentioned, I'm not sure how text is going to work with i18n translations.  One caveat with this is to make sure Protractor tests aren't using the older 'data-test-selector` (for example) attributes if changing them to 'data-test'.
I'm concerned on continuing to support a wide variety of data test attributes, it is going to cause confusion for future developerment.  For example, if another dev picks up where you left off, do you think it's clear which 'data-test...` attribute to use?
This is along the same lines as the many, many css selectors.  I'm wary that they will not be converted to 'data-test` per best practices.  cc @spadgett

Reply:
Also, pls see [Cypress Best Practices: Selecting Elements](https://docs.cypress.io/guides/references/best-practices.html#Selecting-Elements)

Reply:
[/frontend/packages/integration-tests-cypress/README.md](https://github.com/openshift/console/blob/master/frontend/packages/integration-tests-cypress/README.md#best-practices)

Reply:
This 'update to use `data-test` attributes' tech debit is being traced via [JIRA ODC-4937](https://issues.redhat.com/browse/ODC-4937)